packages/block-editor/src/utils/transform-styles/transforms/url-rewrite.js

```diff
@@ -1,138 +1,33 @@
-		meta.value.indexOf( 'data:' ) === 0 ||
-		meta.value.indexOf( '#' ) === 0
-	) {
-		return false;
-	}
-
-	if ( isAbsolutePath( meta.value ) ) {
-		return false;
-	}
-
-	// do not handle the http/https urls if `includeRemote` is false
-	if ( isRemotePath( meta.value ) ) {
-		return false;
-	}
-
-	return true;
-}
-
-/**
- * Get the absolute path of the url, relative to the basePath
- *
- * @param  {string} str          the url
- * @param  {string} baseURL      base URL
- *
- * @return {string}              the full path to the file
- */
-function getResourcePath( str, baseURL ) {
-	return new URL( str, baseURL ).toString();
-}
-
-/**
- * Process the single `url()` pattern
- *
- * @param  {string} baseURL  the base URL for relative URLs
- * @return {Promise}         the Promise
- */
-function processURL( baseURL ) {
-	return ( meta ) => ( {
-		...meta,
-		newUrl:
-			'url(' +
-			meta.before +
-			meta.quote +
-			getResourcePath( meta.value, baseURL ) +
-			meta.quote +
-			meta.after +
-			')',
-	} );
-}
-
-/**
- * Get all `url()`s, and return the meta info
- *
- * @param  {string} value decl.value
- *
- * @return {Array}        the urls
- */
-function getURLs( value ) {
-	const reg = /url\((\s*)(['"]?)(.+?)\2(\s*)\)/g;
-	let match;
-	const URLs = [];
-
-	while ( ( match = reg.exec( value ) ) !== null ) {
-		const meta = {
-			source: match[ 0 ],
-			before: match[ 1 ],
-			quote: match[ 2 ],
-			value: match[ 3 ],
-			after: match[ 4 ],
-		};
-		if ( isValidURL( meta ) ) {
-			URLs.push( meta );
-		}
+const rewrite = ( rootURL ) => ( node ) => {
+	if ( node.type !== 'Url' ) return;
+
+	const urlValue = node.value;
+	let urlValueStr = urlValue.value;
+	let urlQuote = '';
+	if ( urlValue.type === 'String' ) {
+		// String type nodes have the value always wrapped in quotes (like `'` or `"`)
+		const urlValueStrQuoted = urlValue.value;
+		urlQuote = urlValueStrQuoted.charAt( 0 );
+		urlValueStr = trimOutmostChars( urlValueStrQuoted );
 	}
-	return URLs;
-}
 
-/**
- * Replace the raw value's `url()` segment to the new value
- *
- * @param  {string} raw  the raw value
- * @param  {Array}  URLs the URLs to replace
- *
- * @return {string}     the new value
- */
-function replaceURLs( raw, URLs ) {
-	URLs.forEach( ( item ) => {
-		raw = raw.replace( item.source, item.newUrl );
-	} );
+	// bases relative URLs with rootURL
+	const basedUrl = new URL( urlValueStr, rootURL );
 
-	return raw;
-}
+	// skip host-relative, already normalized URLs (e.g. `/images/image.jpg`, without `..`s)
+	if ( basedUrl.pathname === urlValueStr ) return;
 
-const rewrite = ( rootURL ) => ( node ) => {
-	if ( node.type === 'declaration' ) {
-		const updatedURLs = getURLs( node.value ).map( processURL( rootURL ) );
-		return {
-			...node,
-			value: replaceURLs( node.value, updatedURLs ),
-		};
-	}
+	// wraps URL in the originally used quotes (if used)
+	urlValue.value = urlQuote + basedUrl.toString() + urlQuote;
+};
 
-	return node;
+const trimOutmostChars = ( str ) => {
```

Comment:
for regular top-level functions, I think we prefer to avoid arrow functions in general.

Reply:
Yes, I replaced it now with a normal function.